template.rb

```diff
@@ -5,84 +5,52 @@
 RUBY_VERSION = '2.6.5'
 
 def apply_template!
-  # TODO: Recheck if we need this
-  delete_git
   delete_test_folder
 
-  directory 'rails_docker', './', force: true, recursive: false
-  directory 'rails_docker/bin', 'bin', mode: :preserve
-  directory 'shared', './', force: true, recursive: false
-
-  # Setup config
-  remove_file 'config/database.yml'
-  directory 'shared/config', 'config',
-  add_sidekiq_config
-  add_mailer_config
-
-  # Setup Javascripts
-  directory 'shared/app/javascript', 'app/javascript'
-  insert_into_file 'app/javascript/packs/application.js', after: "require\(\"channels\"\)\n" do
-    <<~EOT
-    
-    import 'initializers/';
-    import 'screens/';
-    EOT
-  end
-
-  # Setup Stylesheets
-  remove_file 'app/assets/stylesheets/application.css'
-  directory 'shared/app/assets/stylesheets', 'app/assets/stylesheets'
-
-  # Setup base application controller including the useful concerns
-  directory 'shared/app/controllers', 'app/controllers'
+  template 'Gemfile.tt', force: true
+
+  template 'Dockerfile.tt'
+  template 'docker-compose.dev.yml.tt'
+  template 'docker-compose.test.yml.tt'
+  template 'docker-compose.yml.tt'
+  template '.env.tt'
+  copy_file '.dockerignore'
+
+  copy_file '.eslintignore'
+  copy_file '.eslintrc'
+  copy_file '.flayignore'
+  template '.pronto.yml.tt'
+  copy_file '.pronto_eslint_npm.yml'
+  copy_file '.rubocop.yml'
+  copy_file '.scss-lint.yml'
+  copy_file 'config.reek'
+
+  copy_file '.semaphore.yml'
+  template '.ruby-gemset.tt'
+  template '.ruby-version.tt'
+  copy_file '.editorconfig'
+  copy_file '.npmrc'
+  copy_file 'Procfile'
+  copy_file 'Procfile.dev'
+  template 'README.md.tt', force: true
+
+  apply 'app/template.rb'
+  apply 'bin/template.rb'
+  apply 'config/template.rb'
+
+  apply 'package.json.rb'
+  apply '.gitignore.rb'
 
   remove_turbolinks
 
-  setup_rack_deflater
-  setup_i18n
-  setup_test_env
-  setup_bullet
-  setup_linters
-  setup_gitignore
-
-  # Do not fallback to assets pipeline if a precompiled asset is missed.
-  environment(nil, env: 'test') do
-    <<~EOT
-      # Do not fallback to assets pipeline if a precompiled asset is missed.
-      config.assets.compile = false
-
-    EOT
-  end
-
   after_bundle do
     run 'spring stop'
 
-    # Devise configuration
     generate 'devise:install'
```

Comment:
One note is that this should be a variant later on too as we don't always need Devise. Same as for Webpack below ✌️

Reply:
Acutally for Devise, how about doing this in the main template file:

```
if yes?("Add devise?")
  gem 'devise'

  after_bundle do
    rails_command 'generate devise:install'
    rails_command "generate devise #{model}"
  end
end
```

So in the end, there would be pure two main variants "Web" and "API" and options like Devise, Boostrap, Figaro vs. DotEnv.

Webpack would still come by default with the Web variant.

What do you think?

Reply:
I agree. Later we can add a different folders for that kind of options. 👍 

So for now I am going to leave the variants folder to have only `web` and `api`.